src/lib/meta.js

```diff
@@ -1,91 +1,120 @@
+const { map } = require('d3');
 var escape = require('escape-html'),
-    geojsonRandom = require('geojson-random'),
-    geojsonExtent = require('geojson-extent'),
-    geojsonFlatten = require('geojson-flatten'),
-    polyline = require('polyline'),
-    wkx = require('wkx'),
-    zoomextent = require('../lib/zoomextent');
+  geojsonRandom = require('geojson-random'),
+  geojsonExtent = require('geojson-extent'),
+  geojsonFlatten = require('geojson-flatten'),
+  polyline = require('polyline'),
+  wkx = require('wkx'),
+  zoomextent = require('../lib/zoomextent');
 
 module.exports.adduserlayer = function(context, _url, _name) {
-    var url = escape(_url), name = escape(_name);
-    var layer = L.tileLayer(url, {
-        maxZoom: context.map.getMaxZoom()
+  var url = escape(_url), name = escape(_name);
+
+  // reset the control if a user-layer was added before
+  d3.select('.user-layer-button').remove();
+
+  function addUserSourceAndLayer() {
+    // if the source and layer aren't present, add them
```

Comment:
Would a conditional check wrapping this be useful here?

Reply:
I don't think so, remove will just do nothing if the selection is empty.